packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts

```diff
@@ -0,0 +1,68 @@
+import { Address, ReadOnlyWallet } from '@celo/connect'
+import {
+  bigNumberToBuffer,
+  bufferToBigNumber,
+  getAddressFromPublicKey,
+  publicKeyFromAsn1,
+  publicKeyPrefix,
+  sixtyFour,
+} from '@celo/wallet-hsm'
+import { RemoteWallet } from '@celo/wallet-remote'
+import { KeyManagementServiceClient } from '@google-cloud/kms'
+import { BigNumber } from 'bignumber.js'
+import { GcpHsmSigner } from './gcp-hsm-signer'
+
+/**
+ * A Cloud HSM wallet built on GCP.
+ */
+export class GcpHsmWallet extends RemoteWallet<GcpHsmSigner> implements ReadOnlyWallet {
+  private client: KeyManagementServiceClient | undefined
+
+  constructor(private readonly versionName: string) {
+    super()
+  }
+
+  private generateKmsClient() {
+    return new KeyManagementServiceClient()
+  }
+
+  protected async loadAccountSigners(): Promise<Map<Address, GcpHsmSigner>> {
+    if (!this.client) {
+      this.client = this.generateKmsClient()
+    }
+    const addressToSigner = new Map<Address, GcpHsmSigner>()
+    try {
+      const publicKey = await this.getPublicKeyFromVersionName(this.versionName)
+      const address = getAddressFromPublicKey(publicKey)
+      addressToSigner.set(address, new GcpHsmSigner(this.client, this.versionName, publicKey))
+    } catch (e) {
+      console.error('Error loading account', e)
+      throw e
+    }
+    return addressToSigner
+  }
+
+  private async getPublicKeyFromVersionName(versionName: string): Promise<BigNumber> {
+    if (!this.client) {
+      throw new Error('GcpHsmWallet needs to be initialised first')
+    }
+    const [pk] = await this.client.getPublicKey({ name: versionName })
+    const derEncodedPk = pk.pem?.split('\n').slice(1, -2).join('').trim()
+    // @ts-ignore
+    const pubKey = publicKeyFromAsn1(Buffer.from(derEncodedPk, 'base64'))
```

Comment:
What are you ignoring? Casting `derEncondedPk` was not an option?

Reply:
It looks like pk.pem can be null or undefined. I addressed this by adding an if-statement to throw if this is the case